mllib/src/main/scala/org/apache/spark/ml/param/sharedParams.scala

```diff
@@ -23,6 +23,27 @@ private[ml] trait HasRegParam extends Params {
   def getRegParam: Double = get(regParam)
 }
 
+private[ml] trait HasElasticNetParam extends HasRegParam {
+  /** param for elastic net regularization parameter */
+  val alphaParam: DoubleParam = new DoubleParam(
+    this, "alphaParam", "the ElasticNet mixing parameter")
+  def getAlphaParam: Double = get(alphaParam)
+}
+
+private[ml] trait HasConvergenceTol extends Params {
+  /** param for max number of iterations */
+  val convergenceTol: DoubleParam = new DoubleParam(
+    this, "convergenceTol", "the convergence tolerance for iterative algorithms")
+  def getConvergenceTol: Double = get(convergenceTol)
+}
+
+private[ml] trait HasIntercept extends Params {
```

Comment:
I would like to be able to disable intercept in this PR. Should not be too hard.